packages/validation/src/MonokleValidator.ts

```diff
@@ -152,9 +158,14 @@ export class MonokleValidator {
     return this.#config ?? this.#fallback;
   }
 
-  set config(value: Config | undefined) {
-    this.#config = value;
-    this.cancelLoad("reconfigured");
+  get metadata(): Record<PluginName, PluginMetadataWithConfig> {
```

Comment:
The "big change" is going from an RPC approach (see below validator.hasRule, validator.isRuleEnabled) to a representational state transfer approach. We can now store this exposed metadata in the Redux store and respond like this. This will make it much easier to move the validator completely to a worker instead of relying on synchronous methods.